tests/SprykerSdkTest/Zed/AppSdk/Business/AppAsyncApiFacadeTest.php

```diff
@@ -193,4 +193,28 @@ public function testAddAsyncApiMessageFromTransferObjectWithSpecifiedMessageName
         $this->tester->assertAsyncApiResponseHasNoErrors($asyncApiResponseTransfer);
         $this->tester->assertAsyncApiHasPublishMessageInChannel($asyncApiRequestTransfer->getTargetFile(), 'FooBar', $asyncApiMessageTransfer->getChannel()->getName());
     }
+
+    /**
+     * @return void
+     */
+    public function testAddAsyncApiMessagePropertyAdd(): void
+    {
+        // Arrange
+        $asyncApiMessageTransfer = $this->tester->haveAsyncApiMessagePropertyRequestTransfer();
+
+        $asyncApiRequestTransfer = $this->tester->haveAsyncApiAddRequestWithExistingAsyncApi();
+        $asyncApiRequestTransfer->setAsyncApiMesssage($asyncApiMessageTransfer);
+
+        // Act
+        $asyncApiResponseTransfer = $this->tester->getFacade()->addAsyncApiMessage(
+            $asyncApiRequestTransfer,
+        );
+
+        // Assert
+        $this->tester->assertAsyncApiResponseHasNoErrors($asyncApiResponseTransfer);
+
+        $this->tester->assertChannelProperty($asyncApiRequestTransfer->getTargetFile(), 'schemas', 'message', ['firstName', 'string', true]);
+        $this->tester->assertChannelProperty($asyncApiRequestTransfer->getTargetFile(), 'schemas', 'message', ['lastName', 'string', false]);
+        $this->tester->assertChannelProperty($asyncApiRequestTransfer->getTargetFile(), 'schemas', 'message', ['phoneNumber', 'int', true]);
```

Comment:
does it make sense to pass all properties for assertion as an array and call this function once, or use a data provider?

Reply:
A data provider would make sense but we can change later if needed we need to get it done.

Reply:
I have put properties as data providers.